### ISSUE TITLE
Determine better default `format` for tooltip?

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#4350]
Our `numberFormat` is undefined by default as we're relying on the scales to produce `"nice"` tick labels. 

However, as we also automatically generate tooltips, they currently don't get any formatting by default. 

Since Vega-Lite does not read the data, it is quite difficult to tackle this systematically. 

If we know if the quantitative field contains integers or floats, though some rules could be useful

- If a field contains only integers, a good format is likely `"d"`
- If a field contains floating points / real numbers then keep all the `~r`

**Comment by g3o2** [comment]
Are there any comparable rules for other guides such as `legend` and `axis`?

**Comment by kanitw** [comment]
> Are there any comparable rules for other guides such as legend and axis?

As mentioned, our numberFormat is undefined by default as we're relying on the scales to produce "nice" tick labels.

**Comment by g3o2** [comment]
So this issue should maybe extend to the other guides, too, shouldn't it?

**Comment by kanitw** [comment]
Well, the guides  do not need default format as the tick values are already made nice by the scales.

**Comment by g3o2** [comment]
do I understand correctly that this only holds as long as `nice` is set to `true`?

**Comment by kanitw** [comment]
No, `nice` only adjusts `domain` boundary, but no the tick values. 
(You can try in the editor.)
